src/Moryx.AbstractionLayer.Products.Endpoints/ProductConverter.cs

```diff
@@ -269,22 +269,27 @@ private void UpdateCollection(IList value, IEnumerable<PartModel> parts)
                 if (partModel is null)
                     continue;
 
-                var match = unused.Find(r => r.Id == partModel?.Id);
-                if (match == null)
+                var oldPartMatch = oldParts.Find(r => r.Id == partModel.Id);
+                // new partlink
+                if (oldPartMatch == null)
                 {
-                    match = (IProductPartLink)Activator.CreateInstance(elemType);
-                    value.Add(match);
+                    oldPartMatch = (IProductPartLink)Activator.CreateInstance(elemType);
+                    oldPartMatch.Product = _productManagement.LoadType(partModel.Product.Id);
+                    value.Add(oldPartMatch);
                 }
+                //modified reference
+                else if (oldPartMatch.Product.Id != partModel.Product.Id)
+                    oldPartMatch.Product = _productManagement.LoadType(partModel.Product.Id);
                 else
-                    unused.Remove(match);
+                    //part removed
```

Comment:
better comment: 
// existing unchanged partlink: do not delete at the end

Reply:
I agree